sky/data/storage.py

```diff
@@ -125,7 +130,7 @@ class StoreMetadata:
         def __init__(self,
                      *,
                      name: str,
-                     source: str,
+                     source: Union[Path, List[Path], None],
```

Comment:
Should we define a variable `source_type=Union[Path, List[Path]` and use that everywhere? (Please feel free to use a better name for the variable)

Reply:
Good idea! Applied the changes.